src/devices/Tm1637/Tm1637.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+namespace Iot.Device.Tm1637
+{
+    public class Tm1637 : IDisposable
+    {
+        /// <summary>
+        /// The number of segments that the TM1637 can handle
+        /// </summary>
+        public byte MaxSegments => 6;
+        // According to the doc, the clock pulse width minimum is 400 ns
+        // And waiting time between clk up and down is 1 µs
+        private const byte ClockWidthMicroseconds = 1;
+
+        private readonly int _pinClk;
+        private readonly int _pinDio;
+        private GpioController _controller;
+        private byte _brightness;
+        // Default segment order is from 0 to 5
+        private byte[] _segmentOrder = new byte[6] { 0, 1, 2, 3, 4, 5 };
+        // To store what has been displayed last. Used when change on brightness or
+        // screen on/off is used
+        private byte[] _lastDisplay = new byte[6] { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+        private bool _screenOn;
+
+        /// <summary>
+        /// Initialize a TM1637
+        /// </summary>
+        /// <param name="pinClk">The clock pin</param>
+        /// <param name="pinDio">The data pin</param>
+        /// <param name="pinNumberingScheme">Use the logical or physical pin layout</param>
+        public Tm1637(int pinClk, int pinDio, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical)
```

Comment:
This should also take in an optional IGpioController, so that if one gets passed in, then that one is used for locating the clock and data pin.